src/ark/phenotyping/pixie_preprocessing.py

```diff
@@ -239,6 +257,34 @@ def create_pixel_matrix(fovs, channels, base_dir, tiff_dir, seg_dir,
     if not os.path.exists(os.path.join(base_dir, subset_dir)):
         os.mkdir(os.path.join(base_dir, subset_dir))
 
+    # define path to channel normalization values
+    channel_norm_path = os.path.join(
+        base_dir, pixel_output_dir, 'channel_norm.feather'
+    )
+
+    # define path to pixel normalization values
+    pixel_thresh_path = os.path.join(
+        base_dir, pixel_output_dir, 'pixel_thresh.feather'
+    )
+
+    # reset entire cohort if channels provided are different from ones in existing channel_norm
+    if os.path.exists(channel_norm_path):
+        channel_norm_df = feather.read_dataframe(channel_norm_path)
+
+        if set(channel_norm_df.columns.values) != set(channels):
+            print("New channels provided: overwriting whole cohort")
+
+            # delete the existing data in data_dir and subset_dir
+            rmtree(os.path.join(base_dir, data_dir))
+            os.mkdir(os.path.join(base_dir, data_dir))
+
+            rmtree(os.path.join(base_dir, subset_dir))
+            os.mkdir(os.path.join(base_dir, subset_dir))
+
+            # delete the existing channel_norm.feather and pixel_thresh.feather
+            os.remove(channel_norm_path)
+            os.remove(pixel_thresh_path)
+
     # create variable for storing 99.9% values
     quantile_path = os.path.join(base_dir, data_dir, "channel_norm_post_rowsum_perfov.csv")
```

Comment:
Sorry to be annoying about all these filenames - can we change this to `channel_norm_post_rownorm_perfov.csv'?